src/language-markdown/printer-markdown.js

```diff
@@ -790,9 +790,11 @@ function printUrl(url, dangerousCharOrChars = []) {
       ? dangerousCharOrChars
       : [dangerousCharOrChars]),
   ];
-  return new RegExp(dangerousChars.map((x) => `\\${x}`).join("|")).test(url)
-    ? `<${url}>`
-    : url;
+
+  const newUrl = url.replaceAll("<", "%3C").replaceAll(">", "%3E");
+  return new RegExp(dangerousChars.map((x) => `\\${x}`).join("|")).test(newUrl)
+    ? `<${newUrl}>`
+    : newUrl;
```

Comment:
```suggestion
  return new RegExp(dangerousChars.map((x) => `\\${x}`).join("|")).test(newUrl)
    ? `<${newUrl.replaceAll(">", "%3E")}>`
    :  newUrl;
```

Is this be enough?

Reply:
You are right, escape them both then, but only when we wrap url in `<>`.

Reply:
I'm still new to the project and thought why take the risk down the line so I just escaped for either case. Anyways, I'll just that. Thank you for your help.

Reply:
The original works fine, people may don't want escaped version.

**Prettier pr-15400**
[Playground link](https://deploy-preview-15400--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEBteAPGBdAFACxhgAcBnJAegrgwEMBbYgGzgDpJ6KB+WgXgD4AlCAA0ICMRgBLaKWShaAJ0UQA7gAUlCOSlpNVtAJ5yxAI0W0wAazgwAygzgAZKVDjIAZntJwzF67Z2xJauAObIMIoArr4gPvRSEdGxpGEsAIpREPCe3rEAVqQYdmlwmdnuSF5MPmIAjlnw6ipkyCC0pAC0bnAAJn2iIJG0UkxhAMIQ9PS0bXpMg6lQoSwAgkSKUqZRTXCKLm65NbGE9EwA6vhS8KTBYHB22tdSAG7Xhm1gpCYgLzEAklB+rA7GBNpJVkC7DBDCwjrUQMQVD5zhZiG0kXAfIoXu4xK5sTBmrRQjN4bFgopsW0ZoorL01FBBkjXDBzlJejB8MgABwABjEijgDSkQuJpNmVTyYhgtFM7M53KQACYxFEfAAVOU6aoIuD0Ux9fq9Jy0ZZRElwABiEEUMyIYTmOwgIAAvq6gA)
<!-- prettier-ignore -->
```sh
--parser markdown
```

**Input:**
<!-- prettier-ignore -->
```markdown
[text](https://example.com/?a=>)
```

**Output:**
<!-- prettier-ignore -->
```markdown
[text](https://example.com/?a=%3E)

```

Reply:
Won't this fail when we create a link description: 
[link]: https://www.google.fr/foo->bar

If I don't escape when there is no dangerous character then the > will remain as it is & will break it again.

Reply:
I don't get it, is this what you mean?

**Prettier 3.0.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBtANgSygawLpIAEAFjDAA4DOSA9DQO6MB0A5hBC+nEwGYBONHuwC0APgBGAQz4gANCAjkYmaJWShpfCPQAK0hGpST09SQE8188X0lgccGAGVJAWzgAZbHGQ9jlOFY2dg6O5LbYLMgwfACuASD+LphRsfGUEVwAijEQ8D5+8QBWlAAejhlw2bneSL7o-vIAjjnwOlpUyCCSlMJQcHAAJoNyINGSmFhQLADCEC4ukp3G6CPpU1wAgmR8mOIxrXB8nn359fGkLugA6sSY8JRhYHCOBneYAG53Zp1glJYg7ziAEkoENYI4wDslBtQY4YGYuKcGiByFp-FcbOROqi4P4+O9vPJsHiYG1JCwFkj4mE+HjOgs+DgBtooCNUdgYFdMAMYMRkAAOAAM8j4cGamFFZIpi1qBXkMEk4i5PL5SAATPIYv4ACqKwx1ZFwFziQZDAbuSRTGLkuAAMQgfAWZAiS32EBAAF8PUA)
<!-- prettier-ignore -->
```sh
--parser markdown
```

**Input:**
<!-- prettier-ignore -->
```markdown
[link]: https://www.google.fr/foo->bar
```

**Output:**
<!-- prettier-ignore -->
```markdown
[link]: https://www.google.fr/foo->bar

```

How can we break anything if we don't change it at all.

Reply:
Yes, you are right. It won't. Just got a bit confused.